src/result.rs

```diff
@@ -0,0 +1,103 @@
+use serde_json;
+use std;
+use std::any::Any;
+
+#[derive(Debug)]
+pub enum Error {
+    IO(std::io::Error),
+    JSON(serde_json::Error),
+    AddrParse(std::net::AddrParseError),
+    JoinError(Box<Any + Send + 'static>),
```

Comment:
That seems odd. What's going on here?

Reply:
I am converting the join error type into the result type.  It has a really weird signature because its a return value of the closure that is passed into spawn.

Reply:
That `static most likely comes from the possibility of the thread outliving the parent thread. You can get that out of there by either using an explicit lifetime or using scoped threads from crossbeam.

Reply:
seems like a big change just for the type.  for a return value the 'static' isn't a huge problem because all-most all the apis just return ()